<a href="https://colab.research.google.com/github/planetgitn/Play-Store-App-Review-Analysis/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

There are more than 3.04 million apps found on Google Play Store. With this project we will analyzing various apps found on the play store with the help of different python libraries.

# It consists of 13 columns:- App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, and Android Ver with 10841 Rows.

**Imports:** Let us start by importing some of the required libraries with which we will be working on.

In [3]:
# Imports
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')



# Loading the dataset as a pandas data frame.

In [4]:
googlestore_df =  pd.read_csv('/content/Play Store Data.csv')

In [5]:
googlestore_df.head(10)

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up
5                         Paper flowers instructions  ...    2.3 and up
6            Smoke Effect Photo Maker - Smoke Editor  ...  4.0.3 and up
7                                   Infinite Painter  ...    4.2 and up
8                               Garden Coloring Book  ...    3.0 and up
9                      Kids Paint Free - Drawing Fun  ...  4.0.3 and up

[10 rows x 13 columns]

In [6]:
googlestore_df.tail(10)

App  ...         Android Ver
10831                       payermonstationnement.fr  ...          4.0 and up
10832                                       FR Tides  ...          2.1 and up
10833                                    Chemin (fr)  ...          2.2 and up
10834                                  FR Calculator  ...          4.1 and up
10835                                       FR Forms  ...          4.0 and up
10836                               Sya9a Maroc - FR  ...          4.1 and up
10837               Fr. Mike Schmitz Audio Teachings  ...          4.1 and up
10838                         Parkinson Exercices FR  ...          2.2 and up
10839                  The SCP Foundation DB fr nn5n  ...  Varies with device
10840  iHoroscope - 2018 Daily Horoscope & Astrology  ...  Varies with device

[10 rows x 13 columns]

After loading the dataset, we can start the exploration but before that, we need to check and see that the dataset is ready for performing several exploration operations or not, so let’s first have a look at the structure and the manner in which the data is organized.

# To know if there is any missing value or Nan value in the dataset, we can use the isnull() function.

In [7]:
googlestore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

So, we will need to prepare the dataset before performing exploratory data analysis on it.










We saw that the dataset contains many Null or missing values. The column Rating, Type , Content Rating , Current Ver , andAndroid Ver contains 1474, 1, 1, 8, and 3 missing values respectively.


Will it not be better if we can define a function to get more useful information about the different attributes of the dataset, also there is one more valid point in defining a function which it will be reusable, and we are going to utilize our defined function several times in future


In [8]:
def printinfo():
    temp = pd.DataFrame(index=googlestore_df.columns)
    temp['data_type'] = googlestore_df.dtypes
    temp['null_count'] = googlestore_df.isnull().sum()
    temp['unique_count'] = googlestore_df.nunique()
    return temp

Let’s call the function and see what it returns:

In [9]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33

We have some useful information about the dataset. i.e., we can now see the missing number of values of any attribute, its unique count, and its respective data types.
Now we can start the process of data cleaning, lets start with the column Type :-

In [10]:
googlestore_df[googlestore_df.Type.isnull()]

App  ...         Android Ver
9148  Command & Conquer: Rivals  ...  Varies with device

[1 rows x 13 columns]

In [11]:
googlestore_df['Type'].fillna("Free", inplace = True)

After filling the value we can check and see if that has been correctly placed.

In [12]:
googlestore_df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [14]:

googlestore_df[googlestore_df['Content Rating'].isnull()]

App Category  ...  Current Ver Android Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up         NaN

[1 rows x 13 columns]

In [15]:
googlestore_df.loc[10468:10477, :]

App  ...   Android Ver
10468                         Tassa.fi Finland  ...    4.0 and up
10469              TownWiFi | Wi-Fi Everywhere  ...    4.2 and up
10470                               Jazz Wi-Fi  ...    2.3 and up
10471                         Xposed Wi-Fi-Pwd  ...  4.0.3 and up
10472  Life Made WI-Fi Touchscreen Photo Frame  ...           NaN
10473                  osmino Wi-Fi: free WiFi  ...    4.4 and up
10474                             Sat-Fi Voice  ...    2.2 and up
10475                         Wi-Fi Visualizer  ...    2.3 and up
10476                    Lennox iComfort Wi-Fi  ...  2.3.3 and up
10477              Sci-Fi Sounds and Ringtones  ...    4.0 and up

[10 rows x 13 columns]

We can clearly see that row 10472 has missing data for the Categorycolumn and all the prevailing column values are being replaced with its previous column. A better idea will be to drop this row from our data frame.

In [17]:
googlestore_df.dropna(subset = ['Content Rating'], inplace=True)

We are having some of the unwanted columns which will be of not much use in the analysis process. So let’s drop those columns.

In [18]:
googlestore_df.drop(['Current Ver','Last Updated', 'Android Ver'], axis=1, inplace=True)

In [19]:
modeValueRating = googlestore_df['Rating'].mode()

In [20]:
googlestore_df['Rating'].fillna(value=modeValueRating[0], inplace = True)

Finally, after fixing all the missing values, we should have a look at our data frame, We defined a function as printinfo() . So, it’s time to use that function.


In [21]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews           object           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

All the columns have the null_count as zero, which indicates that now the data frame doesn’t contain any missing values.
Now we are done with the data cleansing part and in a state to start the work for data preparation

In [22]:
googlestore_df['Reviews'] = googlestore_df.Reviews.astype(int)

We can see that the changes have taken its effect or not by calling our printinfo() function.

In [23]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          6001
Size              object           0           461
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

Now, the reviews column has been converted to int type, so now we can move to the Column: Size

In [24]:
googlestore_df['Size'] = googlestore_df.Size.apply(lambda x: x.strip('+'))# Removing the + Sign

Removing the , symbol:

In [25]:
googlestore_df['Size'] = googlestore_df.Size.apply(lambda x: x.replace(',', ''))# For removing the `,`

Replacing the M symbol by multiplying the value with 1000000:

In [26]:
googlestore_df['Size'] = googlestore_df.Size.apply(lambda x: x.replace('M', 'e+6'))# For converting the M to Mega

Replacing the k by multiplying the value with 1000:

In [27]:
googlestore_df['Size'] = googlestore_df.Size.apply(lambda x: x.replace('k', 'e+3'))# For convertinf the K to Kilo

Replacing the Varies with device value with Nan :

In [28]:
googlestore_df['Size'] = googlestore_df.Size.replace('Varies with device', np.NaN)

Now, finally converting all these values to numeric type:

In [29]:
googlestore_df['Size'] = pd.to_numeric(googlestore_df['Size']) # Converting the string to Numeric type

So, after performing all of these operations, we should have a detailed look at that column, so yes again we will call our useful function which we defined. i.e., printinfo()

In [30]:
printinfo()

data_type  null_count  unique_count
App               object           0          9659
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          6001
Size             float64        1695           459
Installs          object           0            21
Type              object           0             2
Price             object           0            92
Content Rating    object           0             6
Genres            object           0           119

we can now convert it from string type to numeric type, and then have a look at our dataset.

In [31]:
googlestore_df.dropna(subset = ['Size'], inplace=True)
googlestore_df['Installs'] = googlestore_df.Installs.apply(lambda x: x.strip('+'))
googlestore_df['Installs'] = googlestore_df.Installs.apply(lambda x: x.replace(',', ''))
googlestore_df['Installs'] = pd.to_numeric(googlestore_df['Installs'])

In [32]:
printinfo()

data_type  null_count  unique_count
App               object           0          8434
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          4680
Size             float64           0           459
Installs           int64           0            20
Type              object           0             2
Price             object           0            87
Content Rating    object           0             6
Genres            object           0           116

So, now we are only left with the Price column.
Column: Price :
Converting this column from objectto Numeric type.

In [33]:
googlestore_df['Price'].value_counts()

0          8421
$0.99       145
$2.99       114
$1.99        66
$4.99        65
           ... 
$1.75         1
$389.99       1
$19.90        1
$74.99        1
$4.80         1
Name: Price, Length: 87, dtype: int64

The values contain a special symbol $ which can be removed and then converted to the numeric type.

In [34]:
googlestore_df['Price'] = googlestore_df.Price.apply(lambda x: x.strip('$'))
googlestore_df['Price'] = pd.to_numeric(googlestore_df['Price'])
printinfo()

data_type  null_count  unique_count
App               object           0          8434
Category          object           0            33
Rating           float64           0            39
Reviews            int64           0          4680
Size             float64           0           459
Installs           int64           0            20
Type              object           0             2
Price            float64           0            87
Content Rating    object           0             6
Genres            object           0           116

# **Exploratory Analysis and Visualization**

Let’s begin by importing matplotlib.pyplot and seaborn , and at the same time set our fig size, font size, etc.

In [35]:
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline

sns.set_style('darkgrid')
matplotlib.rcParams['font.size'] = 14
matplotlib.rcParams['figure.figsize'] = (9, 5)
matplotlib.rcParams['figure.facecolor'] = '#00000000'